document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/parser/filter/MongoStandardNonExprRelationalFilterParser.java

```diff
@@ -16,6 +19,8 @@ public Map<String, Object> parse(
     final String parsedLhs = expression.getLhs().accept(context.lhsParser());
     final String operator = mapping.getOperator(expression.getOperator());
     final Object parsedRhs = expression.getRhs().accept(context.rhsParser());
-    return Map.of(parsedLhs, Map.of(operator, parsedRhs));
+    final Map<String, Object> operatorToRhsMap = new HashMap<>();
```

Comment:
Could you please add a comment for using a "HashMap" here instead of `Map.of()`?

Reply:
done